### ISSUE TITLE
Node does not start on linux arm64

### ISSUE THREAD

**Issue by br0x** [wavesplatform/Waves#1662]
On linux arm aarch64, waves crashes with following log:

$ sudo head -n 48 /var/log/waves/waves.log
```
2018-11-09 19:08:32,166 INFO  [main] c.w.Application$ - Starting...
2018-11-09 19:08:38,261 INFO  [ctor.default-dispatcher-3] a.event.slf4j.Slf4jLogger - Slf4jLogger started
2018-11-09 19:08:38,941 INFO  [main] c.w.Application$ - Waves v0.14.7 Blockchain Id: W
2018-11-09 19:08:38,948 DEBUG [main] c.w.db.package$ - Open DB at /var/lib/waves/data
2018-11-09 19:08:39,384 ERROR [main] c.w.actor.RootActorSystem$ - Error while initializing actor system wavesplatform
java.lang.UnsatisfiedLinkError: Could not load library. Reasons: [no leveldbjni64-1.8 in java.library.path, no leveldbjni-1.8 in java.library.path, no leveldbjni in java.library.path, /tmp/libleveldbjni-64-1-1299760925770483686.8: /tmp/libleveldbjni-64-1-1299760925770483686.8: cannot open shared object file: No such file or directory (Possible cause: architecture word width mismatch)]
	at org.fusesource.hawtjni.runtime.Library.doLoad(Library.java:182)
	at org.fusesource.hawtjni.runtime.Library.load(Library.java:140)
	at org.fusesource.leveldbjni.JniDBFactory.<clinit>(JniDBFactory.java:48)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.wavesplatform.db.LevelDBFactory$.$anonfun$factory$3(LevelDBFactory.scala:16)
	at scala.util.Try$.apply(Try.scala:209)
	at com.wavesplatform.db.LevelDBFactory$.$anonfun$factory$2(LevelDBFactory.scala:16)
	at scala.collection.immutable.List.flatMap(List.scala:335)
	at com.wavesplatform.db.LevelDBFactory$.$anonfun$factory$1(LevelDBFactory.scala:15)
	at scala.collection.TraversableViewLike$FlatMapped.$anonfun$foreach$3$adapted(TraversableViewLike.scala:190)
	at scala.collection.Iterator.foreach(Iterator.scala:944)
	at scala.collection.Iterator.foreach$(Iterator.scala:944)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1432)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.SeqLike$$anon$2.foreach(SeqLike.scala:679)
	at scala.collection.TraversableViewLike$FlatMapped.foreach(TraversableViewLike.scala:189)
	at scala.collection.TraversableViewLike$FlatMapped.foreach$(TraversableViewLike.scala:188)
	at scala.collection.SeqViewLike$$anon$5.foreach(SeqViewLike.scala:197)
	at scala.collection.TraversableViewLike$Transformed.headOption(TraversableViewLike.scala:110)
	at scala.collection.TraversableViewLike$Transformed.headOption$(TraversableViewLike.scala:109)
	at scala.collection.SeqViewLike$AbstractTransformed.headOption(SeqViewLike.scala:37)
	at com.wavesplatform.db.LevelDBFactory$.factory$lzycompute(LevelDBFactory.scala:19)
	at com.wavesplatform.db.LevelDBFactory$.factory(LevelDBFactory.scala:12)
processors: 4
Max memory available: 1908932608
os.name: Linux
os.version: 4.4.132-1075-rockchip-ayufan-ga83beded8524
os.arch: aarch64
java.version: 1.8.0_191
java.vendor: Oracle Corporation
java.home: /usr/lib/jvm/java-8-oracle/jre
```

**Comment by br0x** [comment]
Further investigation shown that leveldbjni jar does not contain proper native artifact for linux-aarch64.
I could build such an artifact, but I need to know exact versions (or git revisions) of google/leveldb, google/snappy, dain/leveldb (aka API version) and fusesource/hawtjni, used to create leveldbjni-all-1.18.3.jar  
I believe someone from ethereum team named 'Stan' created this jar

**Comment by br0x** [comment]
Resolved. I successfully built the jar and tested in qemu aarch64. When I organize my results so they are easily replicable from git sources, I will create the PR

**Comment by Tolsi** [comment]
@br0x can you share the detailed steps of your solution?

**Comment by br0x** [comment]
@Tolsi  Unfortunately I still don't have fully automated script, but for reference I can provide versions I used:

snappy 1.1.3 (Jul 7, 2015). Needed for leveldb
crc32c (latest git rev.). Needed for leveldb 
leveldb 1.20 (git rev. 8143c12f3fc483b1ba61cdce11f9c1faf6d01bea, Mar 16, 2018)
iq80 leveldb 0.6 from maven
hawtjni 1.11 from maven
leveldbjni latest (Nov 1, 2017)

google/leveldb was patched using some patches from here https://github.com/OpenMandrivaAssociation/leveldb

leveldbjni was changed a lot to resolve all build issues

I still have a big fear that currently there is no stable jvm for aarch64 architecture. It often crashes when I run node on real device (I use Rock64 one with 4 cores). Never crashed on QEMU tho...
